catppuccin-sddm: fix custom background substitution

magicquark opened the pull request:
- Upstream changed the default `CustomBackground` configuration option to `true` instead of `false`. Therefore, when `background` is not null the configuration is correct and a substitution is no longer required.

- When `background` is `null`, the configuration remains correct as the default upstream behaviour is to use a default background.

- See: catppuccin/sddm@f5418099#diff-e121ebe3

- Resolves: https://github.com/NixOS/nixpkgs/issues/442758

- Tested with a custom background and the derivation successfully builds and substitutes the custom background path.

- Added a new option `disableBackground` that defaults to `false`. When it is set to `true` it will stop the default transparent overlay background from being loaded.

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

magicquark committed:
catppuccin-sddm: fix custom background substitution

- Upstream changed the default `CustomBackground` configuration option
  to `true` instead of `false`. Therefore, when `background` is not null
  the configuration is correct and a substitution is not required.

- See: catppuccin/sddm@f5418099#diff-e121ebe3

- Resolves: https://github.com/NixOS/nixpkgs/issues/442758

magicquark commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 442829`
Commit: [`bbe3614be270e0497309110b203bb8769d899952`](https://github.com/NixOS/nixpkgs/commit/bbe3614be270e0497309110b203bb8769d899952) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/bbe3614be270e0497309110b203bb8769d899952..pull/442829/head))
Merge: [`9658a0c30398b830996d4fc3d39c823f37b2c1dd`](https://github.com/NixOS/nixpkgs/commit/9658a0c30398b830996d4fc3d39c823f37b2c1dd)

Logs: https://github.com/magicquark/nixpkgs-review-gha/actions/runs/17708044621


---
### `x86_64-linux`
:white_check_mark: *No rebuilds*


---
### `aarch64-linux`
:white_check_mark: *No rebuilds*


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

snorrwe commented:
I think we need to explicitly set `CustomBackground` to `false` if no background is provided, since it defaults to `true` now.

Something like this: https://github.com/NixOS/nixpkgs/pull/443138/files

magicquark committed:
catppuccin-sddm: add `disableBackground` option

- The default upstream behaviour applies a transparent overlay
  background.

- Presently, there is no straightforward way to disable this.

- This commit introduces a `disableBackground` option, defaulting to
  `false` to preserve upstream behaviour.

- When set to `true`, it sets `CustomBackground=false` in the
  configuration file, preventing the background path from being loaded.
